research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,153 @@
+> This document represents a component specification template. Its purpose
+> is to provide a point-in-time guide as to how a component should be
+> authored, in light of common anatomies, behaviors, and accessibility.
+> The primary audience is a developer authoring a given component.
+
+# {ComponentName} component specification
+
+> Enter a description here. Example:
+>
+> The `{ComponentName}` component allows a user to slide a single thumb along a horizontal or vertical axis, representing a min/max range.
+
+## Visual examples
+
+> Provide, if possible, visual examples of the component being built, to provide an idea of implementations.
+
+## DOM hierarchy/structure
+
+> Define the recommended DOM shape to represent the component, and how important attributes (aria attributes especially) are applied to the various parts. In cases where a component nests other components, ideally, you expand the full dom structure to understand both the expectation and if there are shortcommings in the child components' customizability.
+>
+> Example:
+>
+> ```html
+> <div class="root">
+>   <div class="rail"></div>
+>
+>   <div class="mark">
+>     <label class="markLabel" />
+>   </div>
+>   <div class="track" />
+>   <div
+>     class="thumb"
+>     tabindex="0"
+>     role="slider"
+>     aria-valuenow="0"
+>     aria-valuemin="0"
+>     aria-valuemax="10"
+>   />
+>   <input name="{name}" type="hidden" value="0" />
+> </div>
+> ```
+
+## Component anatomy
+
+> Outline the parts of a component; what their names and purposes are. Ideally order them in DOM order. Also consider if the component should be broken into subcomponents. Example:
```

Comment:
In addition to the part definitions, perhaps the template should include a diagram of the anatomy as well? (This could also be categorized under the DOM hierarchy / structure section above). Currently, the WIP specs for <select> and <carousel> both include a diagram.

Reply:
I second the diagram.  Perhaps we can do this in another follow PR so as to get this one merged.

Here's a sample diagram @gregwhitworth shared with me previously that another team/effort was working with:

![image](https://user-images.githubusercontent.com/5067638/69107595-c1856000-0a26-11ea-85b0-7628918f5258.png)

And we also have a `research/carousel` branch here in Open UI that is toying with a CSS Grid approach so that we can write these in our MDX files:

![image](https://user-images.githubusercontent.com/5067638/69107657-f5608580-0a26-11ea-9780-5246b726f4f9.png)

Reply:
We copied the premise of the Carousel variant for the <select> one - it looks like this:

![image](https://user-images.githubusercontent.com/865244/69200923-8d2aa600-0af1-11ea-8bac-98373d06be4d.png)